### ISSUE TITLE
Allow changing the Nominatim endpoint

### ISSUE THREAD

**Issue by enriqueav** [gboeing/osmnx#291]
According to [Nominatim Usage Policy](https://operations.osmfoundation.org/policies/nominatim/) it is not acceptable to use it for heavy loads and/or bulk geocoding.

There are commercial instances of the system, like [MapQuest](https://developer.mapquest.com/documentation/open/nominatim-search/) which offer the same functionality and (in my experience) are a little bit faster.

Osmnx could allow changing the Nominatim API endpoint to use one of these instances. 
Something like

```
ox.utils.config(
    nominatim_endpoint='http://open.mapquestapi.com/nominatim/v1/',
    nominatim_key=my_key
)
```

I'm actually testing this in my fork and it seems to work correctly. If you don't provide a key for a commercial instance, it fails like this:

```
/app/osmnx_source/osmnx/core.py in nominatim_request(params, type, pause_duration, timeout, error_pause_duration)
    295             else:
    296                 log('Server at {} returned status code {} and no JSON data'.format(domain, response.status_code), level=lg.ERROR)
--> 297                 raise Exception('Server returned no JSON data.\n{} {}\n{}'.format(response, response.reason, response.text))
    298 
    299         return response_json

Exception: Server returned no JSON data.
<Response [403]> Forbidden
The AppKey submitted with this request is invalid.
```

Again, if accepted, I could help with the PR.

**Comment by gboeing** [comment]
This could be a useful addition. The challenge would be handling optional URL parameters (such as API keys) in a flexible way. For instance, OSM nominatim expects no API key, but commercial vendors might. We'd need to handle different parameters accordingly.

**Comment by enriqueav** [comment]
https://github.com/gboeing/osmnx/pull/294

**Comment by gboeing** [comment]
closed by #294
